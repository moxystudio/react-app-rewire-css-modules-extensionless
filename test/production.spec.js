'use strict';

const serializerPath = require('jest-serializer-path');
const rewireCssModulesExtensionless = require('..');

expect.addSnapshotSerializer(serializerPath);

const mockConfig = {
    module: {
        rules: [
            {
                test: /\.(js|jsx|mjs)$/,
                enforce: 'pre',
                use: [
                    { options: {}, loader: 'path/to/eslint-loader/index.js' },
                ],
                include: 'path/to/src',
            },
            {
                oneOf: [
                    {
                        test: [/\.bmp$/, /\.gif$/, /\.jpe?g$/, /\.png$/],
                        loader: 'path/to/url-loader/index.js',
                        options: {},
                    },
                    {
                        test: /\.(js|jsx|mjs)$/,
                        include: 'path/to/src',
                        loader: 'path/to/babel-loader/lib/index.js',
                        options: {},
                    },
                    {
                        test: /\.css$/,
                        loader: [
                            {
                                loader: 'path/to/extract-text-webpack-plugin/dist/loader.js',
                                options: {},
                            },
                            {
                                loader: 'path/to/style-loader/index.js',
                                options: {},
                            },
                            {
                                loader: 'path/to/css-loader/index.js',
                                options: {
                                    importLoaders: 1,
                                    minimize: true,
                                    sourceMap: true,
                                },
                            },
                            {
                                loader: 'path/to/postcss-loader/lib/index.js',
                                options: {},
                            },
                        ],
                    },
                    {
                        exclude: [/\.js$/, /\.html$/, /\.json$/],
                        loader: 'path/to/file-loader/dist/cjs.js',
                        options: { name: 'static/media/[name].[hash:8].[ext]' },
                    },
                ],
            },
        ],
    },
};

it('should modify the webpack config correctly', () => {
    const result = rewireCssModulesExtensionless(mockConfig, 'production');

    expect(result.module.rules).toMatchSnapshot();
});

it('should override the default include / exclude', () => {
    const result = rewireCssModulesExtensionless(mockConfig, 'production', {
        include: 'foo',
        exclude: 'bar',
    });

    expect(result.module.rules).toMatchSnapshot();
});

it('should pass options to the css loader', () => {
    const result = rewireCssModulesExtensionless(mockConfig, 'production', {
        camelCase: 'dashes',
        localIdentName: 'foo-[hash:base64:5]!',
    });

    expect(result.module.rules).toMatchSnapshot();
});

it('should allow usage with compose', () => {
    expect(rewireCssModulesExtensionless(mockConfig, 'production'))
    .toEqual(rewireCssModulesExtensionless()(mockConfig, 'production'));

    expect(rewireCssModulesExtensionless(mockConfig, 'production', { camelCase: 'dashes' }))
    .toEqual(rewireCssModulesExtensionless({ camelCase: 'dashes' })(mockConfig, 'production'));
});
