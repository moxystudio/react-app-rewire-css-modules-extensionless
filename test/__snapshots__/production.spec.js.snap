// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should modify the webpack config correctly 1`] = `
Array [
  Object {
    "enforce": "pre",
    "include": "path/to/src",
    "test": /\\\\\\.\\(js\\|jsx\\|mjs\\)\\$/,
    "use": Array [
      Object {
        "loader": "path/to/eslint-loader/index.js",
        "options": Object {},
      },
    ],
  },
  Object {
    "oneOf": Array [
      Object {
        "loader": "path/to/url-loader/index.js",
        "options": Object {},
        "test": Array [
          /\\\\\\.bmp\\$/,
          /\\\\\\.gif\\$/,
          /\\\\\\.jpe\\?g\\$/,
          /\\\\\\.png\\$/,
        ],
      },
      Object {
        "include": "path/to/src",
        "loader": "path/to/babel-loader/lib/index.js",
        "options": Object {},
        "test": /\\\\\\.\\(js\\|jsx\\|mjs\\)\\$/,
      },
      Object {
        "exclude": undefined,
        "include": "<PROJECT_ROOT>/src",
        "loader": Array [
          Object {
            "loader": "path/to/extract-text-webpack-plugin/dist/loader.js",
            "options": Object {},
          },
          Object {
            "loader": "path/to/style-loader/index.js",
            "options": Object {},
          },
          Object {
            "loader": "path/to/css-loader/index.js",
            "options": Object {
              "camelCase": undefined,
              "importLoaders": 1,
              "localIdentName": "[hash:base64:5]!",
              "minimize": true,
              "sourceMap": true,
            },
          },
          Object {
            "loader": "path/to/postcss-loader/lib/index.js",
            "options": Object {},
          },
        ],
        "test": /\\\\\\.css\\$/,
      },
      Object {
        "loader": Array [
          Object {
            "loader": "path/to/extract-text-webpack-plugin/dist/loader.js",
            "options": Object {},
          },
          Object {
            "loader": "path/to/style-loader/index.js",
            "options": Object {},
          },
          Object {
            "loader": "path/to/css-loader/index.js",
            "options": Object {
              "importLoaders": 1,
              "minimize": true,
              "sourceMap": true,
            },
          },
          Object {
            "loader": "path/to/postcss-loader/lib/index.js",
            "options": Object {},
          },
        ],
        "test": /\\\\\\.css\\$/,
      },
      Object {
        "exclude": Array [
          /\\\\\\.js\\$/,
          /\\\\\\.html\\$/,
          /\\\\\\.json\\$/,
        ],
        "loader": "path/to/file-loader/dist/cjs.js",
        "options": Object {
          "name": "static/media/[name].[hash:8].[ext]",
        },
      },
    ],
  },
]
`;

exports[`should override the default include / exclude 1`] = `
Array [
  Object {
    "enforce": "pre",
    "include": "path/to/src",
    "test": /\\\\\\.\\(js\\|jsx\\|mjs\\)\\$/,
    "use": Array [
      Object {
        "loader": "path/to/eslint-loader/index.js",
        "options": Object {},
      },
    ],
  },
  Object {
    "oneOf": Array [
      Object {
        "loader": "path/to/url-loader/index.js",
        "options": Object {},
        "test": Array [
          /\\\\\\.bmp\\$/,
          /\\\\\\.gif\\$/,
          /\\\\\\.jpe\\?g\\$/,
          /\\\\\\.png\\$/,
        ],
      },
      Object {
        "include": "path/to/src",
        "loader": "path/to/babel-loader/lib/index.js",
        "options": Object {},
        "test": /\\\\\\.\\(js\\|jsx\\|mjs\\)\\$/,
      },
      Object {
        "exclude": "bar",
        "include": "foo",
        "loader": Array [
          Object {
            "loader": "path/to/extract-text-webpack-plugin/dist/loader.js",
            "options": Object {},
          },
          Object {
            "loader": "path/to/style-loader/index.js",
            "options": Object {},
          },
          Object {
            "loader": "path/to/css-loader/index.js",
            "options": Object {
              "camelCase": undefined,
              "importLoaders": 1,
              "localIdentName": "[hash:base64:5]!",
              "minimize": true,
              "sourceMap": true,
            },
          },
          Object {
            "loader": "path/to/postcss-loader/lib/index.js",
            "options": Object {},
          },
        ],
        "test": /\\\\\\.css\\$/,
      },
      Object {
        "exclude": undefined,
        "include": "<PROJECT_ROOT>/src",
        "loader": Array [
          Object {
            "loader": "path/to/extract-text-webpack-plugin/dist/loader.js",
            "options": Object {},
          },
          Object {
            "loader": "path/to/style-loader/index.js",
            "options": Object {},
          },
          Object {
            "loader": "path/to/css-loader/index.js",
            "options": Object {
              "camelCase": undefined,
              "importLoaders": 1,
              "localIdentName": "[hash:base64:5]!",
              "minimize": true,
              "sourceMap": true,
            },
          },
          Object {
            "loader": "path/to/postcss-loader/lib/index.js",
            "options": Object {},
          },
        ],
        "test": /\\\\\\.css\\$/,
      },
      Object {
        "loader": Array [
          Object {
            "loader": "path/to/extract-text-webpack-plugin/dist/loader.js",
            "options": Object {},
          },
          Object {
            "loader": "path/to/style-loader/index.js",
            "options": Object {},
          },
          Object {
            "loader": "path/to/css-loader/index.js",
            "options": Object {
              "importLoaders": 1,
              "minimize": true,
              "sourceMap": true,
            },
          },
          Object {
            "loader": "path/to/postcss-loader/lib/index.js",
            "options": Object {},
          },
        ],
        "test": /\\\\\\.css\\$/,
      },
      Object {
        "exclude": Array [
          /\\\\\\.js\\$/,
          /\\\\\\.html\\$/,
          /\\\\\\.json\\$/,
        ],
        "loader": "path/to/file-loader/dist/cjs.js",
        "options": Object {
          "name": "static/media/[name].[hash:8].[ext]",
        },
      },
    ],
  },
]
`;

exports[`should pass options to the css loader 1`] = `
Array [
  Object {
    "enforce": "pre",
    "include": "path/to/src",
    "test": /\\\\\\.\\(js\\|jsx\\|mjs\\)\\$/,
    "use": Array [
      Object {
        "loader": "path/to/eslint-loader/index.js",
        "options": Object {},
      },
    ],
  },
  Object {
    "oneOf": Array [
      Object {
        "loader": "path/to/url-loader/index.js",
        "options": Object {},
        "test": Array [
          /\\\\\\.bmp\\$/,
          /\\\\\\.gif\\$/,
          /\\\\\\.jpe\\?g\\$/,
          /\\\\\\.png\\$/,
        ],
      },
      Object {
        "include": "path/to/src",
        "loader": "path/to/babel-loader/lib/index.js",
        "options": Object {},
        "test": /\\\\\\.\\(js\\|jsx\\|mjs\\)\\$/,
      },
      Object {
        "exclude": undefined,
        "include": "<PROJECT_ROOT>/src",
        "loader": Array [
          Object {
            "loader": "path/to/extract-text-webpack-plugin/dist/loader.js",
            "options": Object {},
          },
          Object {
            "loader": "path/to/style-loader/index.js",
            "options": Object {},
          },
          Object {
            "loader": "path/to/css-loader/index.js",
            "options": Object {
              "camelCase": "dashes",
              "importLoaders": 1,
              "localIdentName": "foo-[hash:base64:5]!",
              "minimize": true,
              "sourceMap": true,
            },
          },
          Object {
            "loader": "path/to/postcss-loader/lib/index.js",
            "options": Object {},
          },
        ],
        "test": /\\\\\\.css\\$/,
      },
      Object {
        "exclude": "bar",
        "include": "foo",
        "loader": Array [
          Object {
            "loader": "path/to/extract-text-webpack-plugin/dist/loader.js",
            "options": Object {},
          },
          Object {
            "loader": "path/to/style-loader/index.js",
            "options": Object {},
          },
          Object {
            "loader": "path/to/css-loader/index.js",
            "options": Object {
              "camelCase": undefined,
              "importLoaders": 1,
              "localIdentName": "[hash:base64:5]!",
              "minimize": true,
              "sourceMap": true,
            },
          },
          Object {
            "loader": "path/to/postcss-loader/lib/index.js",
            "options": Object {},
          },
        ],
        "test": /\\\\\\.css\\$/,
      },
      Object {
        "exclude": undefined,
        "include": "<PROJECT_ROOT>/src",
        "loader": Array [
          Object {
            "loader": "path/to/extract-text-webpack-plugin/dist/loader.js",
            "options": Object {},
          },
          Object {
            "loader": "path/to/style-loader/index.js",
            "options": Object {},
          },
          Object {
            "loader": "path/to/css-loader/index.js",
            "options": Object {
              "camelCase": undefined,
              "importLoaders": 1,
              "localIdentName": "[hash:base64:5]!",
              "minimize": true,
              "sourceMap": true,
            },
          },
          Object {
            "loader": "path/to/postcss-loader/lib/index.js",
            "options": Object {},
          },
        ],
        "test": /\\\\\\.css\\$/,
      },
      Object {
        "loader": Array [
          Object {
            "loader": "path/to/extract-text-webpack-plugin/dist/loader.js",
            "options": Object {},
          },
          Object {
            "loader": "path/to/style-loader/index.js",
            "options": Object {},
          },
          Object {
            "loader": "path/to/css-loader/index.js",
            "options": Object {
              "importLoaders": 1,
              "minimize": true,
              "sourceMap": true,
            },
          },
          Object {
            "loader": "path/to/postcss-loader/lib/index.js",
            "options": Object {},
          },
        ],
        "test": /\\\\\\.css\\$/,
      },
      Object {
        "exclude": Array [
          /\\\\\\.js\\$/,
          /\\\\\\.html\\$/,
          /\\\\\\.json\\$/,
        ],
        "loader": "path/to/file-loader/dist/cjs.js",
        "options": Object {
          "name": "static/media/[name].[hash:8].[ext]",
        },
      },
    ],
  },
]
`;
